test/companion_web/controllers/page_controller_test.exs

```diff
@@ -1,8 +1,17 @@
 defmodule CompanionWeb.PageControllerTest do
   use CompanionWeb.ConnCase
 
-  test "GET /", %{conn: conn} do
+  test "Login required for /", %{conn: conn} do
     conn = get(conn, "/")
+    assert redirected_to(conn, 302) == "/auth/login"
+  end
+
+  test "Logged in shows page" do
+    conn =
+      session_conn()
+      |> put_session(:user, %{email: "foo@example.org", provider: :google})
```

Comment:
We have `provider: :google` here and `provider: "google"` in the only other place we build a session like this. Is the difference meaningful?

Reply:
Probably not, we don't use it anywhere. I wasn't really sure what to store in the session for a logged in user that would be useful, I didn't just want to set a `:user, true`, so I stored the email address and the login provider, in case we wanted to use that in the future.